### PR TITLE
Adding some generic parentheses snippets.

### PR DESCRIPTION
**Pull request by jlconlin**
I've created a few automatic snippets for parentheses (and their cousins). It makes it easy to add a pair of parentheses and put something before, inside, and after it.

### CODE REVIEW

File: UltiSnips/all.snippets [hunk 1]
```diff
@@ -107,4 +107,23 @@ snippet uuid "Random UUID" w
...
+$1[${2:${VISUAL}}]$0
+endsnippet
+
+snippet {} "Brace surround" A
+$1{${2:${VISUAL}}}$0
+endsnippet
+
+snippet <> "Angle surround" A
+$1<${2:${VISUAL}}>$0
+endsnippet
```
Comment (lpil) [inline UltiSnips/all.snippets]: Do these need to be in the ultisnips format? It looks like they could be written in the snipmate format

Comment (jlconlin) [reply UltiSnips/all.snippets]: I suppose they could. I'm not as familiar with the snipmate format. Is it preferred for them to be in the snipmate format?

Comment (lpil) [reply UltiSnips/all.snippets]: Yes, the snipmate format is more widely supported. :)

Comment (jlconlin) [reply UltiSnips/all.snippets]: Does snipmate support automatic trigger like ultisnips? That's an important feature here.

If this was to be done for snipmate, what file would it go under? There isn't an `snippets/all.snippets` file.

Comment (lpil) [reply UltiSnips/all.snippets]: Ah yes! Thanks :)

**Pull request merged**